spec-tests/src/test/java/org/candlepin/spec/VirtSpecTest.java

```diff
@@ -0,0 +1,792 @@
+                ProductAttributes.GuestLimit.withValue("1")));
+        admin.owners().createPool(owner.getKey(), Pools.randomUpstream(virtLimitProduct));
+        Set<ConsumerInstalledProductDTO> installedProducts = toInstalled(virtLimitProduct);
+        updateInstalledProducts(hostClient, host, installedProducts);
+        linkHostToGuests(hostClient, host, GUEST_UUID, OTHER_GUEST_UUID);
+
+        assertThatEntitlementsOf(hostClient, host).isEmpty();
+
+        hostClient.consumers().autoBind(host.getUuid());
+
+        assertThatEntitlementsOf(hostClient, host).isEmpty();
+    }
+
+    @Test
+    public void shouldAutoBindVirtLimitingProductsThatDoCoverGuests() {
+        ProductDTO virtLimitProduct = createVirtLimitProduct("8");
+        admin.owners().createPool(owner.getKey(), Pools.randomUpstream(virtLimitProduct));
+        Set<ConsumerInstalledProductDTO> installedProducts = toInstalled(virtLimitProduct);
+        updateInstalledProducts(hostClient, host, installedProducts);
+        linkHostToGuests(hostClient, host, GUEST_UUID, OTHER_GUEST_UUID);
+
+        assertThatEntitlementsOf(hostClient, host).isEmpty();
+
+        hostClient.consumers().autoBind(host.getUuid());
+
+        List<EntitlementDTO> entitlementAfterBind = hostClient.consumers().listEntitlements(host.getUuid());
+        assertThat(entitlementAfterBind)
+            .hasSize(1)
+            .map(EntitlementDTO::getPool)
+            .map(PoolDTO::getProductId)
+            .containsOnly(virtLimitProduct.getId());
+    }
+
+    @Test
+    public void shouldNotChangeTheQuantityOnSubPoolWhenTheSourceEntitlementQuantityChanges() {
+        linkHostToGuests(hostClient, host, GUEST_UUID);
+        ProductDTO product = admin.ownerProducts()
+            .createProductByOwner(owner.getKey(), Products.withAttributes(
+                ProductAttributes.VirtualLimit.withValue("3"),
+                ProductAttributes.MultiEntitlement.withValue("yes")));
+        admin.owners().createPool(owner.getKey(), Pools.randomUpstream(product));
+
+        List<PoolDTO> createdPools = guestClient.pools()
+            .listPoolsByProduct(owner.getId(), product.getId());
+        assertThat(createdPools).hasSize(2);
+
+        PoolDTO normalPool = findNormalPools(createdPools).stream()
+            .findFirst()
+            .orElseThrow();
+
+        JsonNode entitlement = hostClient.consumers().bindPool(host.getUuid(), normalPool.getId(), 3);
+
+        List<PoolDTO> poolsAfterBind = guestClient.pools()
+            .listPoolsByProduct(owner.getId(), product.getId());
+        assertThat(poolsAfterBind).hasSize(3);
+        assertThat(findNormalPools(poolsAfterBind))
+            .filteredOn(pool -> !normalPool.getId().equals(pool.getId()))
+            .hasSize(1)
+            .map(PoolDTO::getQuantity)
+            .containsOnly(3L);
+
+        hostClient.entitlements()
+            .updateEntitlement(entitlement.get(0).get("id").asText(), toUpdate(entitlement));
```

Comment:
shouldn't this be reducing the quantity from 3 to 2? `toUpdate` it seems to just pass the original entitlement's quantity (3)